packages/aggregator/src/__tests__/analytics/aggregateAnalytics/aggregations/offsetPeriod.test.js

```diff
@@ -4,160 +4,208 @@
  */
 
 import { arrayToAnalytics } from '@tupaia/data-broker';
-import { offsetPeriod } from '../../../../analytics/aggregateAnalytics/aggregations/offsetPeriod';
+import {
+  getDateRangeForOffsetPeriod,
+  offsetPeriod,
+} from '../../../../analytics/aggregateAnalytics/aggregations/offsetPeriod';
 
-describe('offsetPeriod()', () => {
-  it('throws an error if `periodType` is not provided', () => {
-    expect(() => offsetPeriod([], { offset: 1 })).toThrow("'periodType' is required");
-  });
+describe('offsetPeriod', () => {
+  const testConfigValidation = () => {
+    it('throws an error if `periodType` is not provided', () => {
+      expect(() => offsetPeriod([], { offset: 1 })).toThrow("'periodType' is required");
+    });
 
-  it('throws an error if `offset` is not provided', () => {
-    expect(() => offsetPeriod([], { periodType: 'day' })).toThrow("'offset' is required");
-  });
+    it('throws an error if `offset` is not provided', () => {
+      expect(() => offsetPeriod([], { periodType: 'day' })).toThrow("'offset' is required");
+    });
+  };
 
-  it('`periodType` is case insensitive', () => {
-    const analytics = [
-      { dataElement: 'BCD1', organisationUnit: 'TO', period: '20190101', value: 1 },
-    ];
-    const offset = 1;
+  describe('offsetPeriod()', () => {
+    testConfigValidation();
```

Comment:
The only changes here are

1. I created and used the `testConfigValidation` function
2. I wrapped this block in an outer `offsetPeriod` block